packages/example/src/screens/HeadlessCheckoutWithRedirect.tsx

```diff
@@ -29,98 +29,146 @@ const HeadlessCheckoutWithRedirect = (props: any) => {
-        console.log(log);
-        setIsLoading(false);
-        setIsValidating(null);
-      },
-    });
-  };
-
-  const onSubmit = async () => {
+  const submit = async () => {
     try {
       await redirectManager.submit();
     } catch (err) {
       console.error(err);
     }
   };
 
-  const pay = async (id: string) => {
-    try {
-      setIsValidating(id);
-      await redirectManager.onBankSelected(id);
-    } catch (err) {
-      setIsValidating(null);
-      console.error(err);
-    }
-  };
+  return (
+    <View
+      style={{
+        paddingHorizontal: 5,
+        flex: 1,
+        backgroundColor: 'white',
+      }}>
+      <Search
+        search={search}
+        onSearch={async (value: string) => {
+          setSearch(value);
+
+          try {
+            await redirectManager.onBankFilterChange(value);
+          } catch (err) {
+            console.error(err);
+          }
+        }}
+      />
+      <Banks
+        banks={banks}
+        isValidating={isValidating}
+        onPay={async (id: string) => {
+          try {
+            setIsValidating(id);
+            await redirectManager.onBankSelected(id);
+          } catch (err) {
+            setIsValidating(null);
+            console.error(err);
+          }
+        }}
+      />
+      <Loader isLoading={isLoading} />
+    </View>
+  );
+};
+
+
+const Search = ({ search, onSearch }: any) => {
+  return (
+    <TextField
+      title="Choose your bank"
+      textInputStyle={{
+        backgroundColor: '#f5f5f5',
+        borderColor: '#f5f5f5',
+        paddingHorizontal: 10,
+      }}
+      style={{
+        padding: 5,
+        marginVertical: 10,
+        borderRadius: 5,
+        marginHorizontal: 5,
+      }}
+      placeholder="Search Banks"
+      value={search}
+      onChangeText={(value) => {
+        onSearch(value);
+      }}
+    />
+  );
+}
+
+
+const Loader = ({ isLoading }: any) => {
+  if (!isLoading) {
+    return null;
+  } else {
+    return (
+      <View
+        style={{
+          position: 'absolute',
+          left: 0,
+          right: 0,
+          top: 0,
+          bottom: 0,
+          alignItems: 'center',
+          justifyContent: 'center',
+          backgroundColor: 'rgba(200, 200, 200, 0.5)',
+          zIndex: 1000,
+        }}>
+        <ActivityIndicator size="small" />
+      </View>
+    );
+  }
+}
 
-  const renderLoadingOverlay = () => {
-    if (!isLoading) {
-      return null;
-    } else {
-      return (
-        <View
-          style={{
-            position: 'absolute',
-            left: 0,
-            right: 0,
-            top: 0,
-            bottom: 0,
-            alignItems: 'center',
-            justifyContent: 'center',
-            backgroundColor: 'rgba(200, 200, 200, 0.5)',
-            zIndex: 1000,
-          }}>
-          <ActivityIndicator size="small" />
-        </View>
-      );
-    }
-  };
 
+const Banks = ({ banks, isValidating, onPay }: any) => {
```

Comment:
[feedback] please do not use `any` at all, ever.
If the type is not known, use `unknown`.
If the type is known, which is the case in this PR, please type it.
For example, I'm guessing in this line it would be:

```ts
const Banks = ({
  banks,
  isValidating,
  onPay,
}: {
  banks: IBank[];
  isValidating: boolean;
  onPay: (id: string) => void;
}) => {};
```